admin_guide/configuring_authentication.adoc

```diff
@@ -0,0 +1,533 @@
+provider and, if specified, from the JSON returned by the `*UserInfo*` URL.
+
+At least one claim must be configured to use as the user's identity. The
+link:http://openid.net/specs/openid-connect-core-1_0.html#StandardClaims[standard
+identity claim] is `sub`.
+
+You can also indicate which claims to use as the user's preferred user name,
+display name, and email address. If multiple claims are specified, the first one
+with a non-empty value is used. The
+link:http://openid.net/specs/openid-connect-core-1_0.html#StandardClaims[standard
+claims] are:
+
+[horizontal]
+`sub`:: The user identity.
+`preferred_username`:: The preferred user name when provisioning a user.
+`email`:: Email address.
+`name`:: Display name.
+
+NOTE: Using an OpenID Connect identity provider requires users to get a token using `_<master>_/oauth/token/request` to use with command-line tools.
+
+.Standard Master Configuration Using `*OpenIDIdentityProvider*`
+====
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: my_openid_connect <1>
+    challenge: false <2>
+    login: true <3>
+    provider:
+      apiVersion: v1
+      kind: OpenIDIdentityProvider
+      clientID: ... <4>
+      clientSecret: ... <5>
+      claims:
+        id:
+        - sub <6>
+        preferredUsername:
+        - preferred_username
+        name:
+        - name
+        email:
+        - email
+      urls:
+        authorize: https://myidp.example.com/oauth2/authorize <7>
+        token: https://myidp.example.com/oauth2/token <8>
+----
+<1> This provider name is prefixed to the value of the identity claim to form an identity name. It is also used to build the redirect URL.
+<2> `*OpenIDIdentityProvider*` cannot be used to send `WWW-Authenticate` challenges.
+<3> When `true`, unauthenticated token requests from web clients (like the Management Console) are redirected to the authorize URL to log in.
+<4> The client ID of a client registered with the OpenID provider. The client must be allowed to redirect to `_<master>_/oauth2callback/_<identityProviderName>_`.
+<5> The client secret.
+<6> Use the value of the `sub` claim in the returned `id_token` as the user's identity.
+<7> link:http://openid.net/specs/openid-connect-core-1_0.html#AuthorizationEndpoint[Authorization Endpoint] described in the OpenID spec. Must use `https`.
+<8> link:http://openid.net/specs/openid-connect-core-1_0.html#TokenEndpoint[Token Endpoint] described in the OpenID spec. Must use `https`.
+====
+
+A custom certificate bundle, extra scopes, and `*userInfo*` URL can also be specified:
+
+.Full Master Configuration Using `*OpenIDIdentityProvider*`
+====
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: my_openid_connect
+    challenge: false
+    login: true
+    provider:
+      apiVersion: v1
+      kind: OpenIDIdentityProvider
+      clientID: ...
+      clientSecret: ...
+      ca: my-openid-ca-bundle.crt <1>
+      extraScopes: <2>
+      - email
+      - profile
+      claims:
```

Comment:
missing `extraAuthorizeParameters` info added in https://github.com/openshift/openshift-docs/pull/306